Makefile

```diff
@@ -14,8 +14,18 @@ PYODIDE_EMCC=$(PYODIDE_ROOT)/ccache/emcc
 PYODIDE_CXX=$(PYODIDE_ROOT)/ccache/em++
 
 SHELL := /bin/bash
-CC=emcc
-CXX=em++
+
+# 'make', as a performance optimization, may try to parse and run simple shell
+# commands directly rather than launching a subshell for each command.
+# The default Mac make implementation appears to have a bug where it
+# disregards modifications to PATH when performing this optimization.
+# As a result, even though emcc and em++ are added to PATH, Mac make sometimes
+# doesn't find them.
+# To avoid this problem, explicitly find and use the absolute path to
+# emcc and em++.
+CC := $(shell which emcc)
+CXX := $(shell which em++)
```

Comment:
This trick doesn't work on my Mac, running Catalina. It seems like the shell directive doesn't recognize $PATH, so `which emcc` / `which em++` resolves to nothing.

Replacing these with `which emcc` seems to allow compilation to continue, though that feels a little unclean. Not sure what the best thing to do here is.

Reply:
Removing `shell` is probably ok, I suppose. It's probably not inheriting the environment somehow...

Reply:
@wlach hm...  Oh, I wonder:  If you put `shell which emcc` back now without cleaning your repo, does it work now?

As I recall, emcc is downloaded, so it's not actually present in PATH when make is first invoked; only on subsequent invocations.

A hacky workaround might be to replace `:=` with `=`, so the shell command is re-evaluated each time CC is invoked.  A less-hacky approach might be to simply hardcode the relative paths to these binaries?

Reply:
Hardcoding to the relative paths in the build tree seems like a decent option -- since in this case we really want to *always* use the local copies of these in the build tree and don't want to ever use a system-installed emscripten.

Reply:
@aseering no, that doesn't seem to fix it, it still resolves to nothing. `\`which emcc\`` is the only approach that seems to work for me :/

Using the direct paths seems like a good solution to me, though I guess we'll have to make sure that works as expected with ccache.